benchmarks/automated/tests/utils/benchmark.py

```diff
@@ -0,0 +1,382 @@
+                    is_local_execution=self.is_local_execution
+                )
+
+                torchserveHandler.download_workflow_artifacts(
+                    workflow_name, workflow_model_urls, workflow_specfile_url, workflow_handler_url
+                )
+
+                for batch_size in batch_sizes:
+                    # Start torchserve
+                    torchserveHandler.start_torchserve_docker()
+
+                    # Create workflow archive and place in the wf_store
+                    torchserveHandler.create_and_update_workflow_archive(
+                        workflow_name,
+                        os.path.basename(workflow_specfile_url),
+                        os.path.basename(workflow_handler_url),
+                        batch_size,
+                        workers,
+                        batch_delay,
+                        retry_attempts,
+                        timeout_ms,
+                    )
+
+                    # Register
+                    torchserveHandler.register_workflow(url=url)
+
+                    # Run benchmark
+                    apacheBenchHandler.run_apache_bench(
+                        requests=requests,
+                        concurrency=concurrency,
+                        input_file=input_file,
+                        is_workflow=True,
+                        workflow_name=workflow_name,
+                    )
+
+                    # Unregister
+                    torchserveHandler.unregister_workflow(workflow_name=workflow_name)
+
+                    # Stop torchserve
+                    torchserveHandler.stop_torchserve()
+
+                    # Generate report (note: needs to happen after torchserve has stopped)
+                    apacheBenchHandler.generate_report(
+                        requests=requests, concurrency=concurrency, connection=self.connection
+                    )
+
+                    # Move artifacts into a common folder.
+                    remote_artifact_folder = f"/home/ubuntu/{self.benchmark_execution_id}/{self.model_name}/{ec2_instance_type}/{mode}/{batch_size}"
+
+                    self.connection.run(f"mkdir -p {remote_artifact_folder}")
+                    self.connection.run(f"cp -R /home/ubuntu/benchmark/* {remote_artifact_folder}")
+
+                    # Upload artifacts to s3 bucket
+                    self.connection.run(
+                        f"aws s3 cp --recursive /home/ubuntu/{self.benchmark_execution_id}/ {S3_BUCKET_BENCHMARK_ARTIFACTS}/{self.benchmark_execution_id}/"
+                    )
+
+                    time.sleep(3)
+
+                    run(
+                        f"aws s3 cp --recursive /tmp/{self.model_name}/ {S3_BUCKET_BENCHMARK_ARTIFACTS}/{self.benchmark_execution_id}/{self.model_name}/{ec2_instance_type}/{mode}/{batch_size}"
+                    )
+
+                    run(f"rm -rf /tmp/{self.model_name}")
+                    apacheBenchHandler.clean_up()
```

Comment:
The above 3 execute_XXX_benchmark highly duplicated. Please clean up.

Reply:
Done.